kafka-client/src/main/java/dev/responsive/kafka/config/ResponsiveConfig.java

```diff
@@ -65,6 +66,28 @@ public class ResponsiveDriverConfig extends AbstractConfig {
   private static final long REQUEST_TIMEOUT_MS_DEFAULT = 5000L;
 
 
+  // ------------------ performance related configurations --------------------
+
+  // TODO: we should make this configurable per-store
+  public static final String STORAGE_DESIRED_NUM_PARTITION_CONFIG = "responsive.storage.desired.num.partitions";
```

Comment:
Does it make more sense to have this config be the multiplicative factor rather than the resulting partition count, if we may not be able to honor the value passed in here unless it divides evenly?

Reply:
I think it will be impossible to set a good default if it's a multiplicative factor, but we can set a sane default (such as 4096 or something) if it's independent of Kafka partitions. The other thing is that the way it should be computed is `total_kafka_size_mb / 100` - which, again, is independent of number of kafka partitions.

Reply:
I was thinking we would still set a default as the total number of partitions, but only if the user has set this factor, then we would instead use it to compute the desired number. Maybe that's just more confusing 🤷‍♀️